### PR TITLE
Enable azure tests

### PR DESCRIPTION
**Pull request by fao89**
[noissue]

Please be sure you have read our documentation on creating PRs:
https://docs.pulpproject.org/contributing/pull-request-walkthrough.html

### CODE REVIEW

**Comment by daviddavis** [comment]
Change LGTM but it seems the test is failing.

**Comment by fao89** [comment]
> Change LGTM but it seems the test is failing.

yep, I'll take a look on what is breaking azure

**Comment by fao89** [comment]
Not sure if it is Azure or Azurite, but it doesn't raise error:
```python
   ...: f1f-b79c-b26c5bd40f22/1.iso"

In [4]: response = requests.get(url, headers={"x-ms-range": "bytes=10985860-10985870"})

In [5]: response.status_code
Out[5]: 206

In [6]: response.headers
Out[6]: {'Server': 'Azurite-Blob/3.14.2', 'last-modified': 'Fri, 01 Oct 2021 19:10:01 GMT', 'x-ms-creation-time': 'Fri, 01 Oct 2021 19:02:06 GMT', 'content-length': '-500100', 'content-type': 'application/octet-stream', 'content-range': 'bytes 10985860-10485759/10485760', 'etag': '"0x1CFD59B66D49660"', 'content-md5': '1B2M2Y8AsgTpgAmY7PhCfg==', 'x-ms-blob-type': 'BlockBlob', 'x-ms-lease-state': 'available', 'x-ms-lease-status': 'unlocked', 'x-ms-request-id': '96f22075-4264-4613-80ee-85d63f4dcbe0', 'x-ms-version': '2020-10-02', 'accept-ranges': 'bytes', 'date': 'Fri, 01 Oct 2021 19:10:44 GMT', 'x-ms-server-encrypted': 'true', 'Connection': 'keep-alive', 'Keep-Alive': 'timeout=5'}

In [7]: response = requests.get(url, headers={"Range": "bytes=10985860-10985870"})

In [8]: response.status_code
Out[8]: 206

In [9]: response.headers
Out[9]: {'Server': 'Azurite-Blob/3.14.2', 'last-modified': 'Fri, 01 Oct 2021 19:10:01 GMT', 'x-ms-creation-time': 'Fri, 01 Oct 2021 19:02:06 GMT', 'content-length': '-500100', 'content-type': 'application/octet-stream', 'content-range': 'bytes 10985860-10485759/10485760', 'etag': '"0x1CFD59B66D49660"', 'content-md5': '1B2M2Y8AsgTpgAmY7PhCfg==', 'x-ms-blob-type': 'BlockBlob', 'x-ms-lease-state': 'available', 'x-ms-lease-status': 'unlocked', 'x-ms-request-id': 'cb50348d-e00a-4f7d-8167-6fa54b7027fd', 'x-ms-version': '2020-10-02', 'accept-ranges': 'bytes', 'date': 'Fri, 01 Oct 2021 19:11:21 GMT', 'x-ms-server-encrypted': 'true', 'Connection': 'keep-alive', 'Keep-Alive': 'timeout=5'}
```

**Comment by fao89** [comment]
yep, the issue is definitely on azurite: https://github.com/Azure/Azurite/blob/main/src/blob/handlers/BlobHandler.ts#L1102-L1105

File: pulpcore/tests/functional/api/using_plugin/test_distributions.py [hunk 1]
```diff
@@ -383,6 +384,10 @@ def test_content_served_immediate_with_range_request_too_large_end_value(self):
 
     def test_content_served_immediate_with_range_request_start_value_larger_than_content(self):
         """Assert that a range request with a start value larger than the content errors."""
+        if "azure" == os.getenv("TEST"):
```
Comment (mdellweg) [inline pulpcore/tests/functional/api/using_plugin/test_distributions.py:387]: Is the behavior on "real" azure the same? Then we should make this assertion based on the default storage class.

Comment (fao89) [reply pulpcore/tests/functional/api/using_plugin/test_distributions.py:387]: I don't have an azure account to confirm, but from the documentation, it seems azure handles it correctly:
https://docs.microsoft.com/en-us/rest/api/storageservices/specifying-the-range-header-for-blob-service-operations
I believe it only happens on azurite, and I opened this PR:
https://github.com/Azure/Azurite/pull/1077

**Review by mdellweg — APPROVED** [review]

**Review by dkliban — APPROVED** [review]

**Pull request merged**